coachmark/src/main/java/com/pseudoankit/coachmark/overlay/OverlayLayout.kt

```diff
@@ -0,0 +1,147 @@
+package com.pseudoankit.coachmark.overlay
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.UiComposable
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.layout.Measurable
+import androidx.compose.ui.layout.Placeable
+import androidx.compose.ui.layout.layoutId
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.pseudoankit.coachmark.model.ToolTipPlacement
+import com.pseudoankit.coachmark.model.TooltipConfig
+
+/** Containing composable must use these values for layoutId on current and previous tooltip. */
+public enum class OverlayChildLayoutId { CURRENT, PREVIOUS }
+
+/** Proposed minimum useful tooltip width; this can be adjusted if needed */
+private const val TOOLTIP_MAX_WIDTH_OVERRIDE_PX = 30
+
+/**
+ * Extracted from DimOverlayEffect, in case another overlay effect needs it in the future.
+ *
+ * @param configCurrent of coach mark highlight; null suits case where there's no "current" tooltip in content
+ * @param configPrevious of coach mark highlight; null suits case where there's no "previous" tooltip in content
+ * @param gapTooltipScreen min distance between tooltip and edge of screen
+ */
+@Composable
+public fun OverlayLayout(
+    content: @Composable @UiComposable () -> Unit,
+    configCurrent: TooltipConfig?,
+    configPrevious: TooltipConfig?,
+    modifier: Modifier = Modifier,
+    gapTooltipScreen: Dp = 8.dp,
+) {
+    Layout(content, modifier) { measurables, constraints ->
+
+        // child count < 2 occurs on first and last coach mark
+        require(measurables.size <= 2) { "OverlayLayout cannot have more than two children" }
```

Comment:
should be fine for now, but if usecases increase do we really need this error?

Reply:
A check like this saved me some frustration once, when I went back and added another child composable that then never showed up in the UI.  But I'll remove it unless you tell me otherwise.

Reply:
cool, works for now can figure out if something comes up later